runtime/lua/lsp/client.lua

```diff
@@ -0,0 +1,304 @@
+local uv = require('luv')
```

Comment:
Is luv already included by default?

For me, inside of nvim:

```
:lua print(package.searchpath('luv', package.cpath))
/home/aktau/github/neovim/neovim/.deps/usr/lib/lua/5.1/luv.so
```

```
:lua print(package.cpath)
./?.so;/usr/local/lib/lua/5.1/?.so;/home/aktau/github/neovim/neovim/.deps/usr/lib/lua/5.1/?.so;/usr/local/lib/
lua/5.1/loadall.so
```

Or:

```
$ nvim -u NONE +'lua io.stderr:write(package.cpath)' +'lua os.exit()'
./?.so;/usr/local/lib/lua/5.1/?.so;/home/aktau/github/neovim/neovim/.deps/usr/lib/lua/5.1/?.so;/usr/local/lib/lua/5.1/loadall.so%                                                                                                            
```

It appears my build folder is in my `package.cpath`, this is the default package path from the LuaJIT compiled into nvim, but it can be overridden if the user sets their own path. This should be made much more robust. It is also not guaranteed that a user has the neovim .deps folder on their machine. We should attempt to not rely on optional libraries (especially those that need to be compiled, like luv).